### PR TITLE
Introduce dex_aggregator_base_trades

### PR DESCRIPTION
**Pull request by Hosuke**
Background: #6215

### CODE REVIEW

File: dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_trades.sql [hunk 1]
```diff
@@ -30,37 +30,96 @@
...
+        {{ model }}
+    {% if is_incremental() %}
+    WHERE {{ incremental_predicate('block_time') }}
+    {% endif %}
+    {% if not loop.last %}
+    UNION ALL
+    {% endif %}
+    {% endfor %}
+)
+                                                                                    en
```
Comment (Hosuke) [inline dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_trades.sql:86]:
```suggestion

```

**Comment by Hosuke** [comment]
I will try to refactor `lifi.trades` in this PR and test if the new workflow works.

**Comment by Hosuke** [comment]
[dex_aggregator.trades check](https://dune.com/queries/4040829)

File: dbt_subprojects/dex/models/_projects/lifi/lifi_base_trades.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{{ config(
+        schema = 'lifi',
+        alias = 'base_trades',
+        post_hook='{{ expose_spells(blockchains = \'["fantom", "optimism"]\',
+                                    spell_type = "project",
+                                    spell_name = "lifi",
+                                    contributors = \'["Henrystats", "hosuke"]\') }}'
```
Comment (jeff-dude) [inline dbt_subprojects/dex/models/_projects/lifi/lifi_base_trades.sql:7]:
```suggestion
        alias = 'base_trades'
```

File: dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_base_trades.sql [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            , block_date
+            , block_month
+            , block_time
+            , token_bought_symbol
+            , token_sold_symbol
+            , token_pair
+            , token_bought_amount
+            , token_sold_amount
+            , try_cast(token_bought_amount_raw as uint256) as token_bought_amount_raw
+            , try_cast(token_sold_amount_raw as uint256) as token_sold_amount_raw
```
Comment (jeff-dude) [inline dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_base_trades.sql:35]:
```suggestion
            , cast(token_bought_amount_raw as uint256) as token_bought_amount_raw
            , cast(token_sold_amount_raw as uint256) as token_sold_amount_raw
```

File: dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_trades.sql [hunk 4]
```diff
@@ -30,37 +28,94 @@
...
-{% endif %}
+WITH aggregator_base_trades AS (
+    SELECT *
+    FROM {{ ref('dex_aggregator_base_trades') }}
+    {% if is_incremental() %}
+    WHERE {{ incremental_predicate('block_time') }}
+    {% endif %}
+)
+
+, enrriched_aggregator_base_trades AS (
```
Comment (jeff-dude) [inline dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_trades.sql:39]:
```suggestion
, enriched_aggregator_base_trades AS (
```

File: dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_trades.sql [hunk 5]
```diff
@@ -30,37 +28,94 @@
...
+    {% if is_incremental() %}
+    WHERE {{ incremental_predicate('block_time') }}
+    {% endif %}
+    {% if not loop.last %}
+    UNION ALL
+    {% endif %}
+    {% endfor %}
+)
+{% set cte_to_union = [
+    'enrriched_aggregator_base_trades'
```
Comment (jeff-dude) [inline dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_trades.sql:85]:
```suggestion
    'enriched_aggregator_base_trades'
```

File: dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_trades.sql [hunk 6]
```diff
@@ -30,37 +28,94 @@
...
+        , block_date
+        , block_month
+        , block_time
+        , token_bought_symbol
+        , token_sold_symbol
+        , token_pair
+        , token_bought_amount
+        , token_sold_amount
+        , try_cast(token_bought_amount_raw as uint256) as token_bought_amount_raw
+        , try_cast(token_sold_amount_raw as uint256) as token_sold_amount_raw
```
Comment (jeff-dude) [inline dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_trades.sql:62]:
```suggestion
        , cast(token_bought_amount_raw as uint256) as token_bought_amount_raw
        , cast(token_sold_amount_raw as uint256) as token_sold_amount_raw
```

File: dbt_subprojects/dex/models/_projects/lifi/optimism/lifi_v2_optimism_trades.sql [hunk 7]
```diff
@@ -101,23 +96,3 @@ left join {{ source('tokens', 'erc20') }} erc20a
 left join {{ source('tokens', 'erc20') }} erc20b
     on erc20b.contract_address = dexs.token_sold_address
     and erc20b.blockchain = 'optimism'
```
Comment (jeff-dude) [inline dbt_subprojects/dex/models/_projects/lifi/optimism/lifi_v2_optimism_trades.sql:98]:
```suggestion
left join {{ source('tokens', 'erc20') }} erc20b
    on erc20b.contract_address = dexs.token_sold_address
    and erc20b.blockchain = 'optimism'
```

while we're here, should we remove erc20 metadata too, and apply at the end with prices? similar to dex?

**Review by jeff-dude — CHANGES_REQUESTED** [review]
left initial feedback below

File: dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_base_trades.sql [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+            , project
+            , version
+            , block_date
+            , block_month
+            , block_time
+            , token_bought_symbol
+            , token_sold_symbol
+            , token_pair
+            , token_bought_amount
+            , token_sold_amount
```
Comment (Hosuke) [inline dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_base_trades.sql:33]: I will try to remove these 5 columns from base_trades in order to remove `erc20` source table from base_trades spells.

**Comment by Hosuke** [comment]
[checkpoint](https://dune.com/queries/4043824)

File: dbt_subprojects/dex/macros/models/enrich_dex_aggregator_trades.sql [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+)
+
+SELECT
+    blockchain
+    , project
+    , version
+    , block_month
+    , block_date
+    , block_time
+--     , block_number
```
Comment (Hosuke) [inline dbt_subprojects/dex/macros/models/enrich_dex_aggregator_trades.sql:77]: `block_number` may be supported in future.

File: dbt_subprojects/dex/models/_projects/lifi/optimism/lifi_v2_optimism_trades.sql (hunk 7 above)
Comment (Hosuke) [reply dbt_subprojects/dex/models/_projects/lifi/optimism/lifi_v2_optimism_trades.sql]: I drafted a similar enrichment macro in this PR.

**Pull request merged**